pcsx2/R5900.cpp

```diff
@@ -548,6 +549,7 @@ void __fastcall eeloadHook()
 
 	wxString discelf;
 	int disctype = GetPS2ElfName(discelf);
+	psxMode = disctype == 1;
```

Comment:
This isn't the most elegant PSX mode detector but for now it's good enough. (It might misdetect modes when starting a homebrew .elf with a PSX disc mounted.)

Reply:
*TODO: elgant-ify detection*?

Reply:
Do you guys have any thoughts as to a better place to hook in? So far this is the best thing I've found; `GetPS2ElfName` returns 2 for a PS2 disc, 1 for a PSX disc, 0 for all others. `mcd->IsPSX()` isn't any good because PS2 games still probe PSX cards.

Reply:
If rama said that's good, I have nothing else to add. 
If it's still not 'perfect' though, a note just for the records would be cool.

Reply:
There's some hardware register that indicates PSX mode with a flag. Once set, it can never be unset, unless doing a hardware reset. It's somewhere in my mess of reversing notes :p